crates/core/plugin_sm/src/plugin_manager.rs

```diff
@@ -188,7 +192,11 @@ impl ExternalPlugins {
 
                 if let Some(file_name) = path.file_name() {
                     if let Some(plugin_name) = file_name.to_str() {
-                        let plugin = ExternalPluginCommand::new(plugin_name, &path);
+                        let plugin = ExternalPluginCommand::new(
+                            plugin_name,
+                            &path,
+                            self.config_location.clone(),
```

Comment:
It would be better to read the config outside the loop `for maybe_entry in fs::read_dir(&self.plugin_dir)` and to pass the value of `config.software.plugin.max_packages` to each new `ExternalPluginCommand`.